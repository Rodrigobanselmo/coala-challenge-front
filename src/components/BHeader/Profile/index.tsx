// import { Avatar, Box, Flex, Text, useBreakpointValue } from '@chakra-ui/react';

import Avatar from "@mui/material/Avatar";
import Box from "@mui/material/Box";
import Text from "@mui/material/Typography";

import { IProfileProps } from "./types";
import { useAuth } from "../../../core/contexts/AuthContext";

export function Profile({}: IProfileProps): JSX.Element {
  const { user } = useAuth();

  const name = user?.name || "";
  const email = user?.email || "";

  return (
    <Box style={{ alignItems: "center", display: "flex", cursor: "pointer" }}>
      {name && (
        <Box mr={8} textAlign="right">
          <Text color={"text.main"} sx={{ fontSize: "0.99rem" }}>
            {name}
          </Text>
          <Text
            color={"text.main"}
            mt={-1}
            sx={{ opacity: 0.5, fontSize: "0.81rem" }}
          >
            {email}
          </Text>
        </Box>
      )}
      <Avatar
        src={user?.photoUrl || "/icons/profile.svg"}
        alt={name}
        sx={{
          backgroundColor: "gray.700",
          width: ["32px", "48px"],
          height: ["32px", "48px"],
          "& .MuiAvatar-img": {
            backgroundColor: "background.default",
            objectFit: "cover",
          },
        }}
      >
        {name.split(" ")[0][0]}
        {name.split(" ")[1]?.[0] || ""}
      </Avatar>
    </Box>
  );
}
