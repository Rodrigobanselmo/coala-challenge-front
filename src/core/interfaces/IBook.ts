export type IBook = {
  id: number;
  title: string;
  description: string;
  authors?: string[];
  categories?: string[];
  pageCount?: number;
  googleId?: string;
  publishedAt?: Date;
  language?: string;
  smallThumbnail?: string;
  thumbnail?: string;
  updatedAt: Date;
  createdAt: Date;
  deletedAt?: Date;
};
