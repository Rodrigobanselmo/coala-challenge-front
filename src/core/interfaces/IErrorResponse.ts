import { IconProps } from "@mui/material";

export interface IErrorResponse {
  response: {
    status: number;
    data: {
      message: string;
    };
  };
}
