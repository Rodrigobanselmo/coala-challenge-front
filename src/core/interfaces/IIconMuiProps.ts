import { IconProps } from "@mui/material";

export interface IIconMuiProps {
  muiName?: string;
  fontSize?: IconProps["fontSize"];
  color?: IconProps["color"];
}
