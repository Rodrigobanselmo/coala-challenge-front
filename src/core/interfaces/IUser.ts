export type IUser = {
  id: string;
  email: string;
  name: string;
  updatedAt: Date;
  createdAt: Date;
  deletedAt?: Date | null;
  photoUrl: string | null;
};
