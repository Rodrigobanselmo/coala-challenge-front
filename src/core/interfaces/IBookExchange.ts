import { StatusEnum } from "../enums/status.enum";
import { IBook } from "./IBook";
import { IUser } from "./IUser";

export type IBookExchange = {
  id: number;
  interestBookId: number;
  targetUserId: string;
  askingBookId: number;
  interetUserId: string;
  updatedAt: Date;
  createdAt: Date;
  deletedAt: Date | null;
  interestBook?: IBook;
  targetUser?: IUser;
  askingBook?: IBook;
  interetUser?: IUser;
  status: StatusEnum;
};
