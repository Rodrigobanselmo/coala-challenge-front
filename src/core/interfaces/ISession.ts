import { IUser } from "./IUser";

export type ISession = {
  token: string;
  user: IUser;
};
