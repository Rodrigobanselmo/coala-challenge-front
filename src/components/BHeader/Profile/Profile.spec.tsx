import React from "react";
import { render } from "@testing-library/react";
import { Profile } from "./Profile";
import { IUser } from "../../../core/interfaces/IUser";

describe("Profile", () => {
  it("matches the snapshot", () => {
    const user: IUser = {
      name: "John Doe",
      email: "john@example.com",
      photoUrl: "/path/to/photo.png",
      id: "1",
      createdAt: new Date(),
      updatedAt: new Date(),
    };

    const { container } = render(<Profile user={user} />);
    expect(container).toMatchSnapshot();
  });

  it("matches the snapshot when user is null", () => {
    const { container } = render(<Profile user={null} />);
    expect(container).toMatchSnapshot();
  });
});
