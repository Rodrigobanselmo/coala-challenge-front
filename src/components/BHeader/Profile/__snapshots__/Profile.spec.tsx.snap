// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile matches the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    style="align-items: center; display: flex; cursor: pointer;"
  >
    <div
      class="MuiBox-root css-1aoryhx"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1a8qzm-MuiTypography-root"
      >
        John Doe
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-148lykm-MuiTypography-root"
      >
        john@example.com
      </p>
    </div>
    <div
      class="MuiAvatar-root MuiAvatar-circular css-8v8enj-MuiAvatar-root"
      data-testid="BAvatar"
    >
      <img
        alt="John Doe"
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/path/to/photo.png"
      />
    </div>
  </div>
</div>
`;

exports[`Profile matches the snapshot when user is null 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
    style="align-items: center; display: flex; cursor: pointer;"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular css-8v8enj-MuiAvatar-root"
      data-testid="BAvatar"
    >
      <img
        alt=""
        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
        src="/icons/profile.svg"
      />
    </div>
  </div>
</div>
`;
